src/main/groovy/com/github/spotbugs/snom/SpotBugsTask.groovy

```diff
@@ -175,7 +178,7 @@ class SpotBugsTask extends DefaultTask implements VerificationTask {
      * Property to specify the name of project. Some reporting formats use this property.
      * Default value is {@code "${project.name} (${task.name})"}.
      */
-    @Input
+    @Internal
```

Comment:
Why this change is necessary? It seems needless for me. 🤔

Reply:
By marking property `projectName` as an input, this has the ability to cause cache misses. I understand this value might show up in the output reports, but it has the ability to affect the cache. Marking as `@Internal` means it is not considered as part of the master cache key for tasks of type `SpotBugsTask`.

The test on this PR will definitely fail if we revert this line.

Another possibility is that we do not change this annotation.  However projects will potentially have a lowered cache rate - especially projects who do not explicitly set `rootProject.name = '...'` in their settings.gradle file. There should be documentation in this case saying that the default value for `projectName` can cause low cache hit %.

Reply:
Another possibility is the `@Console` annotation: https://docs.gradle.org/current/javadoc/org/gradle/api/tasks/Console.html

Reply:
Thanks! I understood.

I'm little bit confused because `projectName` is actually input of this task, and it affects not only console but also report file. So even though annotating with `@Internal` or `@Console` actually solves this problem, I feel little strange to use (not so semantic?).

I'll request author to add a comment to use `@Internal`, for future code reviewers.